binarylog/binarylog_end2end_test.go

```diff
@@ -1059,3 +1061,39 @@ func (s) TestServerBinaryLogFullDuplexError(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+// TestCanceledStatus ensures a server that responds with a Canceled status has
+// its trailers logged appropriately and is not treated as a canceled RPC.
+func (s) TestCanceledStatus(t *testing.T) {
```

Comment:
This tests not hitting one of the three conditionals that cause cancel to be logged in finish(), and cause trailers to be logged. Are the other three conditionals that trigger a cancel to be logged tested already?

Reply:
There is an existing cancel test here: https://github.com/grpc/grpc-go/blob/7f66074c371e4718407f94251720f8e12f211cb7/binarylog/binarylog_end2end_test.go#L961

Reply:
Sounds good.